core/src/test/kotlin/org/evomaster/core/config/ConfigPathCheckTest.kt

```diff
@@ -0,0 +1,155 @@
+
+        filePathRoot.createDirectory()
+
+        // set the file to read only
+        filePathRoot.setReadOnly()
+
+
+        val exception = Assertions.assertThrows(
+            IllegalArgumentException::class.java
+        ) {
+            EMConfig.validateOptions(args)
+        }
+
+        Assertions.assertEquals(exception.message, "Parameter 'outputFolder' refers to a file that does not exist, " +
+                "but the provided file path cannot be used to create a directory: $absoluteFilePath\n" +
+                "Please check file permissions of parent directories")
+
+
+        // delete path to write tests
+        filePathRoot.deleteRecursively()
+
+    }
+
+    /**
+     * The directory to write tests exists, but it is read-only.
+     */
+    @Test
+    fun checkExistingDirectoryReadOnly() {
+        val args = arrayOf("--outputFolder", pathToWriteTests)
+
+        val filePathRoot = File("path1")
+        val file = File(pathToWriteTests)
+        val absolutePath = file.absolutePath
+
+        if (file.exists()) {
+            throw FileAlreadyExistsException(
+                "File with name: path1 already exists " +
+                        "in the current working directory"
+            )
+        }
+
+        // create directory and set it read only
+        file.createDirectory()
+        file.setReadOnly()
+
+        val exception = Assertions.assertThrows(
+            IllegalArgumentException::class.java
+        ) {
+            EMConfig.validateOptions(args)
+        }
+
+        Assertions.assertEquals(exception.message, "Parameter 'outputFolder' refers to a folder that already " +
+                "exists, but that cannot be written to: $absolutePath")
+
+        // delete path to write tests
+        filePathRoot.deleteRecursively()
+
+    }
+
+    /**
+     * The path to the directory to write tests is a file, not a folder.
+     */
+    @Test
+    fun checkExistingFileNotDirectoryReadOnly() {
+        val args = arrayOf("--outputFolder", pathToWriteTests)
+
+        val filePathRoot = File("path1")
+        val file = File(pathToWriteTests)
+        val absolutePath = file.absolutePath
+
+        if (file.exists()) {
+            throw FileAlreadyExistsException(
+                "File with name: path1 already exists " +
+                        "in the current working directory"
+            )
+        }
+
+        // create directories for path2
+        Files.createDirectories(Paths.get("path1/path2"))
+
+        // create path3 as a new file
+        file.createNewFile()
+
+        val exception = Assertions.assertThrows(
+            IllegalArgumentException::class.java
+        ) {
+            EMConfig.validateOptions(args)
+        }
+
+        Assertions.assertEquals(exception.message, "Parameter 'outputFolder' refers to a file that already" +
+                " exists, but that it is not a folder: $absolutePath")
+
+        // delete path to write tests
+        filePathRoot.deleteRecursively()
```

Comment:
same comment as before

Reply:
Hi Andrea,

Thanks for your comment. I have applied this comment to all test cases.